Services/SwissBitcoinPayService.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System.Net;
+using System.Net.Http;
+using System.Security.Cryptography;
+using System.Text;
+using Newtonsoft.Json;
+using SmartStore.SwissBitcoinPay.Models;
+using SmartStore.SwissBitcoinPay.Settings;
+
+namespace SmartStore.SwissBitcoinPay.Services
+{
+    public class SwissBitcoinPayService
+    {
+
+        public static bool CheckSecretKey(string key, string message, string signature)
+        {
+            var msgBytes = HMACSHA256.HashData(Encoding.UTF8.GetBytes(key), Encoding.UTF8.GetBytes(message));
+            string hashString = string.Empty;
+            foreach (byte x in msgBytes)
+            {
+                hashString += String.Format("{0:x2}", x);
+            }
+            return (hashString == signature);
+        }
+
+        public string CreateInvoice(SwissBitcoinPaySettings settings, PaymentDataModel paymentData)
+        {
+
+            try
+            {
+                var invoice = new SwissBitcoinPayInvoiceModel()
+				{
+                    title = paymentData.Description,
+                    description = $"{paymentData.BuyerName}#Order:{paymentData.OrderID}#Store:{paymentData.StoreID.ToString()}",
+                    unit = paymentData.CurrencyCode,
+					amount = paymentData.Amount,
+                    email = paymentData.BuyerEmail,
+                    emailLanguage = paymentData.Lang,
+                    redirectAfterPaid = paymentData.RedirectionURL,
+                    webhook = paymentData.WebHookURL,
+                    delay = 60,
+                    onChain = settings.AcceptOnChain

```

Comment:
Lastly I'd like to include a `customNote` so the merchant have a a distinguishable ID when seeing the list of his orders :

```suggestion
                    onChain = settings.AcceptOnChain,
                    extra = {
                      customNote: $"Order {paymentData.OrderID}"
                    }
```
(not sure about the synthax in C# to put a sub-value in an `extra` object)

![image](https://github.com/SwissBitcoinPay/smartstore-plugin/assets/128961268/ee15ceaa-0c45-47bc-8dbd-caaa55f370d2)

Reply:
I don't see an extra field in the API documentation ?

Reply:
It's not documented yet, but it is implemented. You can use it.